lib/battle/CBattleInfoCallback.cpp

```diff
@@ -855,6 +855,24 @@ AccessibilityInfo CBattleInfoCallback::getAccesibility() const
 		ret[BattleHex(0, y)] = EAccessibility::SIDE_COLUMN;
 	}
 
+	//special battlefields with logically unavailable tiles
+	std::vector<BattleHex> impassableHexes;
+	if(battleGetBattlefieldType().num == BFieldType::SHIP_TO_SHIP)
+		impassableHexes = 
+		{ 

```

Comment:
1)If there are brackets inside, outside brackets required. 
2) Your first example is questionable.